tests/data_context/test_data_context.py

```diff
@@ -1788,6 +1788,112 @@ def test_get_validator_with_attach_expectation_suite(
     )
     assert my_validator.expectation_suite_name == "A_expectation_suite"
 
+def test_get_validator_without_expectation_suite(
+    empty_data_context_stats_enabled, tmp_path_factory
+):
+    context: DataContext = empty_data_context_stats_enabled
+
+    base_directory = str(
+        tmp_path_factory.mktemp(
+            "test_get_validator_without_expectation_suite"
+        )
+    )
+
+    create_files_in_directory(
+        directory=base_directory,
+        file_name_list=[
+            "some_file.csv",
+        ],
+    )
+
+    yaml_config = f"""
+class_name: Datasource
+
+execution_engine:
+    class_name: PandasExecutionEngine
+
+data_connectors:
+    my_filesystem_data_connector:
+        class_name: ConfiguredAssetFilesystemDataConnector
+        base_directory: {base_directory}
+        default_regex:
+            pattern: (.+)\\.csv
+            group_names:
+                - alphanumeric
+        assets:
+            A:
+"""
+
+    config = yaml.load(yaml_config)
+    context.add_datasource(
+        "my_directory_datasource",
+        **config,
+    )
+
+    my_validator = context.get_validator(
+        datasource_name="my_directory_datasource",
+        data_connector_name="my_filesystem_data_connector",
+        data_asset_name="A",
+        batch_identifiers={
+            "alphanumeric": "some_file",
+        },
+    )
+    assert type(my_validator.get_expectation_suite()) == ExpectationSuite
+    assert my_validator.expectation_suite_name == "default"
+
+def test_get_validator_with_batch(
+    empty_data_context, tmp_path_factory
+):
+    context = empty_data_context
```

Comment:
Same general note about context - I think we could probably use `BaseDataContext` and mock any I/O.